### PR TITLE
 implemented task and tests

### PR DESCRIPTION
**Pull request by ivadimko**

### CODE REVIEW

File: src/getAverage.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * Implement a custom array method getAverage:
+ *
+ * Method works in different ways:
+ * - without arguments: calculates average value in array (sum of elements
+ * divided by array length)
+ *
+ * - with string argument: calculates average value of objects in array where
+ * property is a given string
```
Comment (mgrinko) [inline src/getAverage.js]: I don't understand the second condition

Comment (mgrinko) [reply src/getAverage.js]: @roman-mate please help us with english here

Comment (mgrinko) [reply src/getAverage.js]: * With no args it returns an average value of `items` in the array
* With string argument it returns an average value of `item[prop]`
* With function - it returns an average value of function results
* Functional argument is called with 3 args - an item, index, and an array itself

Comment (roman-mate) [reply src/getAverage.js]: positional arguments which have different meaning based on argument type is an anti-pattern. Let's use options instead. 

https://stackoverflow.com/questions/12826977/multiple-arguments-vs-options-object

File: src/getAverage.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+/**
+ * Implement a custom array method getAverage:
```
Comment (roman-mate) [inline src/getAverage.js]: suggestion: function names should start with the verb. Maybe calculateAverage?

File: src/getAverage.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * Method works in different ways:
```
Comment (roman-mate) [inline src/getAverage.js]: Function calculates the average value based on passed options. 
1. if options are not passed, returns average value of all items in the array
2. if option.propertyName is passed, returns average value of `propertyName` property of all items in the array.
3. if option.accumulator = function(item, index, array) is passed, returns average value of accumulator result applied to all items in the array.

File: src/getAverage.js (hunk 2 above)
Comment (ivadimko) [reply src/getAverage.js]: agree

File: src/getAverage.js (hunk 3 above)
Comment (ivadimko) [reply src/getAverage.js]: will change passed argument to `options` with this description

Comment (ivadimko) [reply src/getAverage.js]: @roman-mate @mgrinko should we pass `array` argument to the accumulator function? We don't have tests to check it.

Comment (ivadimko) [reply src/getAverage.js]: And what is the behavior of `array` argument here?
```
users.calculateAverage({
  accumulator: (user, index, array) => user.age + (index * 100)) === 130
}
```
We're not able to change the array option here like in other array methods. Should we just check `array === users` in this case?

Comment (roman-mate) [reply src/getAverage.js]: yeah, since we have item and index, array feels useless. maybe let's not pass it.

File: src/getAverage.js (hunk 1 above)
Comment (mgrinko) [reply src/getAverage.js]: It is not about positional arguments but about [Function Polymorphism](https://www.oreilly.com/library/view/programming-javascript-applications/9781491950289/ch02.html)(search for 'Function Polymorphism' on the page please)

I wanted then to work with `typeof` operator

File: src/getAverage.js (hunk 2 above)
Comment (mgrinko) [reply src/getAverage.js]: `get` is a verb
But I agree that here we have calculation rather than simple getting

File: src/getAverage.js (hunk 3 above)
Comment (mgrinko) [reply src/getAverage.js]: @roman-mate Good description, thanks!

File: src/getAverage.js (hunk 1 above)
Comment (roman-mate) [reply src/getAverage.js]: It's a good idea to teach them typeof. 

Let's add 
* if options.accumulator is not a function, returns undefined.

And add a test case when we pass options.accumulator which is not a function. Then they practice typeof but we don't teach them antipatterns.

File: src/getAverage.js (hunk 3 above)
Comment (mgrinko) [reply src/getAverage.js]: I added `array` as a third argument as it is natural for all the iteration methods
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#Parameters

Comment (mgrinko) [reply src/getAverage.js]: It is very easy to test

File: src/getAverage.js (hunk 1 above)
Comment (ivadimko) [reply src/getAverage.js]: Guys! @roman-mate @danmysak @mgrinko 
```
 * [1, 2, 6].calculateAverage() === 3
 *
 * [1, 2, 6].calculateAverage({propertyName: 'name'}) === undefined
 *
 * const users = [
 *  { id: 1, age: 10 },
 *  { id: 2, age: 20 },
 *  { id: 3, age: 60 },
 * ]
 *
 * users.calculateAverage({propertyName: 'id'}) === 2
 * users.calculateAverage({propertyName: 'name'}) === undefined
 * users.calculateAverage({
 *  accumulator: (user, index) => user.age + (index * 100)
 * }) === 130
 * users.calculateAverage({ accumulator: 'hello world' }) === undefined
```
wdyt?

Comment (roman-mate) [reply src/getAverage.js]: like it!

File: src/getAverage.js (hunk 3 above)
Comment (roman-mate) [reply src/getAverage.js]: good point. then let's keep it.

File: src/getAverage.js (hunk 1 above)
Comment (ivadimko) [reply src/getAverage.js]: Guys! @roman-mate @mgrinko @danmysak 
Will add also these test cases. I let them accept only `object` as an argument
```
 * [1, 2, 6].calculateAverage(['options', 'as', 'array']) === undefined
 * 
 * [1, 2, 6].calculateAverage(null) === undefined
 * 
 * [1, 2, 6].calculateAverage('options as string') === undefined
```

Comment (ivadimko) [reply src/getAverage.js]: also they should return `undefined` for empty object
```
* [1, 2, 6].calculateAverage({}) === undefined
```

Comment (mgrinko) [reply src/getAverage.js]: I wanted to keep it simple
That Why I think this extra conditions will add more confusion than profit

Comment (ivadimko) [reply src/getAverage.js]: let's mark it with *
It's a really interesting task. Btw I've implemented test cases so I don't want to lose the value :)

**Pull request merged**